test/basic.js

```diff
@@ -445,15 +445,17 @@ tape('writes are batched', function (t) {
   var ws = feed.createWriteStream()
 
   ws.write('ab')
-  ws.write('cd')
-  ws.write('ef')
-  ws.end(function () {
-    t.deepEquals(trackingRam.log.data, [
-      { write: [0, Buffer.from('ab')] },
-      { write: [2, Buffer.from('cdef')] }
-    ])
-    feed.close(function () {
-      t.end()
+  setImmediate(function () {
+    ws.write('cd')
+    ws.write('ef')
+    ws.end(function () {
+      t.deepEquals(trackingRam.log.data, [
+        { write: [0, Buffer.from('ab')] },
+        { write: [2, Buffer.from('cdef')] }
+      ])
+      feed.close(function () {
+        t.end()
+      })
```

Comment:
would rather change the result of the assert, ie that only one write is performed and then not have the setImmediate.

thats a cool new feature we get from streamx :)

Reply:
Hmm, i think its a good idea to test blocking still (if eventually a block will be formed), is it a good idea to have a test that has two instantly and two after the setImmediate?

Reply:
yea that makes sense

Reply:
fixed in https://github.com/hypercore-protocol/hypercore/pull/276/commits/875889d140848a090f8f8bbe199672f1481d77d2